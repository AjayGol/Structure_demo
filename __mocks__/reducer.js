import { combineReducers } from 'redux';
import AccountReducer from '../src/modules/account/reducer';
import BrokerReducer from '../src/modules/broker/reducer';
import HomeReducer from '../src/modules/home/reducer';
import SearchReducer from '../src/modules/search/reducer';
import WatchlistReducer from '../src/modules/watchlist/reducer';
import OrdersReducer from '../src/modules/watchlist/reducer';
import NotificationReducer from '../src/modules/notification/reducer';
import buyAssetReducer from '../src/modules/buy-assets/reducer';

export default combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});
