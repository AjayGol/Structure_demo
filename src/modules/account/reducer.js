import {
  SET_SAFE_AREA_INTENT,
  API_WAITING_TIME,
  START_LOADING,
  SET_USER_DETAIL,
  CUSTOM_POPUP,
  EVENT_DATA,
  SET_USER_OTHER_DETAIL,
} from './constant';

export const initStateAccount = {
  email: process.env.NODE_ENV === 'development' ? 'ketan@gmail.com' : '',
  password: process.env.NODE_ENV === 'development' ? 'test1234' : '',
  userName: '',
  token: '',
  safeAreaInsetsData: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDataChat: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDefault: { top: 0, bottom: 0, left: 0, right: 0 },
  apiWaitingTime: 0,
  isLoading: false,
  customPopUp: { isShow: false, data: {} },
  userDetails: {
    id: '',
    name: '',
    username: '',
    mobile: '',
    password: '',
    birthday: '',
    gender: '',
    address: [],
    tradingFrequency: 1,
    brokers: [],
    brokerLoginDetails: {},
    tradingFamiliarity: [],
    traderProgress: {
      traderReturns: 0,
      followerReturns: 0,
      AUM: 0,
      followedPortfolioCount: 0,
      _id: '',
      trader: '',
      id: '',
      twitterUserData: {
        isLoggedIn: false,
      },
    },
    userDetailsOtherData: {
      watchlistsLength: 0,
      ifFirstTradeExecuted: false,
    },
    profileCompletionRatio: {
      profile: 0,
      twitter: 0,
      pan: 0,
      brokers: 0,
      knowledge: 0,
    },
    eventData: false,
  },
};

export default (state = initStateAccount, action) => {
  switch (action.type) {
    case SET_SAFE_AREA_INTENT: {
      return {
        ...state,
        safeAreaInsetsDefault: action.payload,
        safeAreaInsetsData: action.payload,
      };
    }
    case START_LOADING: {
      return {
        ...state,
        isLoading: action.payload,
      };
    }
    case API_WAITING_TIME: {
      return {
        ...state,
        apiWaitingTime: action.payload,
      };
    }
    case SET_USER_DETAIL: {
      return {
        ...state,
        userDetails: action.payload,
        userDetailsOtherData: action.payloadAll,
      };
    }
    case SET_USER_OTHER_DETAIL: {
      return {
        ...state,
        userDetailsOtherData: action.payload,
      };
    }
    case CUSTOM_POPUP: {
      return {
        ...state,
        customPopUp: action.payload,
      };
    }
    case EVENT_DATA: {
      return {
        ...state,
        eventData: action.payload,
      };
    }

    default:
      return state;
  }
};
