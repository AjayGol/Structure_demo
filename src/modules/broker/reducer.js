import { ANGEL_URL } from './constant';

export const initStateBroker = {
  angelURL: '',
};

export default (state = initStateBroker, action) => {
  switch (action.type) {
    case ANGEL_URL: {
      return {
        ...state,
        angelURL: action.payload,
      };
    }

    default:
      return state;
  }
};
