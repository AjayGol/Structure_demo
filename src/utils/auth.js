export const GOOGLE_SIGN_IN_CONFIG = {
  webClientId:
    '500383754555-6249knhcfu3mojs0vgkmnjk4v0tcri9f.apps.googleusercontent.com',
  offlineAccess: true,
};

export const SUPPORT_DIGITS_PLUGIN = false;
