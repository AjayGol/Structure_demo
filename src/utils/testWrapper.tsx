// @ts-ignore
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../../__mocks__/reducer';
const store = createStore(reducer);

function render(
  ui: React.ReactElement<
    any,
    | string
    | ((
        props: any,
      ) => React.ReactElement<
        any,
        string | any | (new (props: any) => React.Component<any, any, any>)
      > | null)
    | (new (props: any) => React.Component<any, any, any>)
  >,
  { locale = 'en', ...renderOptions } = {},
) {
  const Wrapper: React.FC = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

// override React Testing Library's render with our own
export { render };
