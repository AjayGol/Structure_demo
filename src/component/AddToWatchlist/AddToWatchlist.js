import React, { useState, useEffect } from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image,
  ActivityIndicator,
} from 'react-native';
import constant from '../../helper/constant';
import fonts from '../../helper/fonts';
import Animated, { Easing } from 'react-native-reanimated';
import { strLocale } from 'locale';
import { connect } from 'react-redux';
import TextInputCustom from '../../component/TextInputCustom/TextInputCustom';
import { filterGroup } from '../../modules/home/validator';
import { setLoader } from '../../modules/account/actions';
import PickerAlert from '../Picker/Picker';
import { addPortfolioToWatchlistAction } from '../../modules/watchlist/actions';
import { updateWatchlistData } from '../../modules/home/actions';
import { UserEvent } from '../../helper/fabricHelper/track';

const CustomAlert = props => {
  const [name, setName] = useState({
    value: props.title || '',
    error: '',
    focus: false,
  });
  const [pickerShow, setPickerShow] = useState(false);

  const { textValue, listDropdown } = props;
  //
  const {
    container,
    btnContainer,
    btnText,
    btnTextLeft,
    btnContainerLeft,
    btnContainerRight,
    titleText,
    buttonContainer,
    cancelButton,
    cancel,
  } = styles;

  const onPickerSelect = (name, index) => {
    props.onDropDown(index - 1);
  };

  const onPressPickers = () => {
    setPickerShow(true);
    setTimeout(() => {
      setPickerShow(false);
    }, 100);
    UserEvent.userTrackScreen('portfolio_watch_selectwatchlist');
  };

  const renderBottomSection = () => {
    return (
      <View style={buttonContainer}>
        <TouchableOpacity
          style={[btnContainer, btnContainerLeft]}
          onPress={() => props.onAlertClick()}>
          <Text style={[btnTextLeft]}>{strLocale('Cancel')}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          testID={'portfolio.save'}
          style={[btnContainer, btnContainerRight]}
          onPress={() => props.onAlertClick('save')}>
          {(props.isLoading && <ActivityIndicator color={'#FFF'} />) || (
            <Text style={btnText}>
              {(props.portFolioId === '' && strLocale('Add')) ||
                strLocale('Save')}
            </Text>
          )}
        </TouchableOpacity>
      </View>
    );
  };

  return (
    <View style={[container]}>
      <Text style={titleText}>
        {strLocale('home.Add portfolio to watchlist')}
      </Text>
      <TouchableOpacity
        testID={'Add portfolio to watchlist'}
        onPress={onPressPickers}>
        <View style={{ position: 'absolute', bottom: 30, right: 150 }}>
          <PickerAlert
            key={101}
            pickerShow={pickerShow}
            data={listDropdown}
            onPickerSelect={onPickerSelect}
          />
        </View>
        <TextInputCustom
          title={strLocale('home.WATCHLIST')}
          // placeHolder={textValue}
          titleStyle={{ marginTop: 32 }}
          validation={{ value: textValue, error: '' }}
          disable={false}
          setValue={(text, isCheck = false) => {
            setName({ value: text, error: '' });
          }}
        />
      </TouchableOpacity>
      {renderBottomSection()}
      <TouchableOpacity
        onPress={() => props.onAlertClick()}
        style={cancelButton}>
        <Image resizeMode="contain" style={cancel} source={{ uri: 'cancel' }} />
      </TouchableOpacity>
    </View>
  );
};

const AddToWatchList = props => {
  const { data } = props;

  const [fadeAnim] = useState(new Animated.Value(0));
  const [filter, setFilter] = useState(
    (props.data && props.data.type) || filterGroup(),
  );
  const [isError, setIsError] = useState(false);
  const [isPublic] = useState((data && data.isPublic) || false);
  const [title] = useState((data && data.title) || '');
  const [selectedValue, setSelectedValue] = useState(0);
  const [listDropdown, setListDropdown] = useState([]);

  useEffect(() => {
    setTimeout(() => {
      Animated.timing(fadeAnim, {
        duration: 200,
        toValue: 1,
        easing: Easing.out(Easing.linear),
      }).start();
    }, 5);

    let output = [];
    props.watchlist.forEach(({ name, input }) => {
      output.push(name);
    });
    setListDropdown(output);
  }, []);

  const onDropDown = value => {
    setSelectedValue(value);
  };

  const onAlertClick = (title, typeSelected = false) => {
    if (title === 'save') {
      props.setLoader(true);
      props
        .addPortfolioToWatchlistAction(
          props.watchlist[selectedValue].id,
          data.id,
          true,
        )
        .then(res => {
          UserEvent.userTrackScreen('portfolio_watch_add');
          props.setLoader(false);
          updateData();
          hideAlert(title, typeSelected, data.id);
          try {
            props.onToWatchList();
          } catch (e) {}
        })
        .catch(err => {
          props.setLoader(false);
        });
    } else {
      UserEvent.userTrackScreen('portfolio_watch_cancel');
      hideAlert(title, typeSelected);
    }
  };

  const updateData = () => {
    props.updateWatchlistData(data.id, props.watchlist[selectedValue].id);
  };

  const hideAlert = (title, typeSelected, value) => {
    setTimeout(() => {
      Animated.timing(fadeAnim, {
        toValue: 0,
        duration: 100,
        easing: Easing.out(Easing.linear),
      }).start(() => {
        props.onAlertClick(value);
      });
    }, 50);
  };

  const onCreatePortfolio = value => {};

  const onUpdateIsError = () => {
    setIsError(false);
  };

  return (
    <Animated.View
      style={[
        styles.alertOuter,
        { opacity: fadeAnim },
        !!constant.isiPAD && { alignItems: 'center' },
      ]}>
      <CustomAlert
        data={props.filter}
        onAlertClick={onAlertClick}
        inputAlertData={props.inputAlertData}
        isInputAlert={props.isInputAlert}
        manageRenameInputValue={props.manageRenameInputValue}
        dataPortfolio={filter}
        onCreatePortfolio={onCreatePortfolio}
        isError={isError}
        updateIsError={onUpdateIsError}
        isLoading={props.isLoading}
        title={title}
        isPublic={isPublic}
        portFolioId={(data && data.id) || ''}
        textValue={
          (props.watchlist !== 0 && props.watchlist[selectedValue].name) ||
          'Default'
        }
        listDropdown={listDropdown}
        onDropDown={onDropDown}
      />
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 6,
    backgroundColor: '#FFFFFF',
    paddingTop: 31,
    paddingBottom: 25,
    paddingHorizontal: 23,
  },
  alertOuter: {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    position: 'absolute',
    paddingHorizontal: 11,
    backgroundColor: 'rgba(5,34,61,0.4)',
    justifyContent: 'center',
    zIndex: 111111,
    elevation: 3,
  },
  titleText: {
    fontFamily: fonts.fontRubikRegular,
    fontSize: 23,
    color: constant.appGrayColor,
    marginRight: 33,
  },
  btnContainer: {
    height: 56,
    borderRadius: 8,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: constant.appBlueColor,
  },
  btnContainerLeft: {
    backgroundColor: '#FFFFFF',
    marginRight: 5,
  },
  btnContainerRight: {
    marginLeft: 5,
  },
  btnTextLeft: {
    color: constant.appBlueColor,
    fontFamily: fonts.fontViga,
    fontSize: 16,
  },
  btnText: {
    color: '#fff',
    fontFamily: fonts.fontViga,
    fontSize: 16,
  },
  isIpad: {
    maxWidth: 400,
    width: '100%',
  },
  buttonContainer: {
    flexDirection: 'row',
    marginTop: 52,
  },
  cancelButton: {
    position: 'absolute',
    top: 12,
    right: 12,
    width: 21,
    height: 21,
  },
  cancel: {
    width: 21,
    height: 21,
  },
});

const mapStateToProps = state => {
  return {
    safeAreaInsetsDefault: state.account.safeAreaInsetsDefault,
    isLoading: state.account.isLoading,
    watchlist: state.watchlist.watchlists,
  };
};

export default connect(mapStateToProps, {
  setLoader,
  addPortfolioToWatchlistAction,
  updateWatchlistData,
})(AddToWatchList);
