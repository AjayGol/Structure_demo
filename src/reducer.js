import { combineReducers } from 'redux';
import AccountReducer from './modules/account/reducer';
import HomeReducer from './modules/home/reducer';
import SearchReducer from './modules/search/reducer';
import BrokerReducer from './modules/broker/reducer';
import NotificationReducer from './modules/notification/reducer';
import { initStateAccount } from './modules/account/reducer';
import { initStateHome } from './modules/home/reducer';
import WatchlistReducer from './modules/watchlist/reducer';
import { initStateWatchlist } from './modules/watchlist/reducer';
import { initStateOrders } from './modules/orders/reducer';
import OrdersReducer from './modules/orders/reducer';
import { initStateSearch } from './modules/search/reducer';
import { initStateBroker } from './modules/broker/reducer';
import { initStateNotification } from './modules/notification/reducer';
import { initStateBuyAsset } from './modules/buy-assets/reducer';
import buyAssetReducer from './modules/buy-assets/reducer';

export const appDefaultReducer = {
  account: initStateAccount,
  home: initStateHome,
  orders: initStateOrders,
  search: initStateSearch,
  broker: initStateBroker,
  watchlist: initStateWatchlist,
  buyAsset: initStateBuyAsset,
  notification: initStateNotification,
};

const appReducer = combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});

export default function rootReducer(state, action) {
  let finalState = appReducer(state, action);
  if (action.type === 'RESET_STORE') {
    finalState = appDefaultReducer; //resetReducer(finalState, action);
  }
  return finalState;
}
