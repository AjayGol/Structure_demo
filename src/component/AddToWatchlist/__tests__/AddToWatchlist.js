import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import reducer from '../../../../__mocks__/reducer';
import { Provider } from 'react-redux';
import AddToWatchlist from '../AddToWatchlist';
import { render } from '../../../utils/testWrapper';
import TextInputCustom from '../../TextInputCustom/TextInputCustom';
const store = createStore(reducer);

let tree;
let title = 'AddToWatchlist';
beforeEach(() => {
  tree = shallow(
    <Provider store={store}>
      <AddToWatchlist />
    </Provider>,
  );
});

test(`${title} - renders correctly`, () => {
  expect(tree).toMatchSnapshot();
});

it(`${title} - loading test`, () => {
  const screen = render(<AddToWatchlist />);
  screen.getByTestId('Add portfolio to watchlist');
  // expect(tree.find(TextInputCustom));
});

it(`${title} - not loading test`, () => {
  const screen = render(<AddToWatchlist twitterWebURL={'test'} />);
  screen.getByTestId('WebView');
});
