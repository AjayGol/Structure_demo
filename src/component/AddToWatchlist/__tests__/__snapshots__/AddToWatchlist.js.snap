// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToWatchlist - renders correctly 1`] = `ShallowWrapper {}`;
